hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -1082,4 +1090,23 @@ private void completeObservables(Set<String> observables, Throwable error) {
             }
         }
     }
+
+    /**
+     * Register a membership listener which checks if this member is
+     * master when a member is removed and start scanning for jobs.
+     */
+    private void registerMasterListener() {
+        nodeEngine.getClusterService().addMembershipListener(new MembershipListener() {
+            @Override
+            public void memberAdded(MembershipEvent membershipEvent) {
+            }
+
+            @Override
+            public void memberRemoved(MembershipEvent membershipEvent) {
+                if (isMaster()) {
```

Comment:
I am not familiar with how the master is chosen. If a master is removed, is the new master known at the time when this method is called?

Reply:
This will be called when there are change in cluster topology related to master, even when there are no Jet jobs. I think this is what happens in the `ProxySplitBrainTest`.